### PR TITLE
Adds the ability to bind a key to swap groups among screens.

### PR DESCRIPTION
**Pull request by kovasap**
For example, if you have dual monitors and you want the group on one monitor to
switch to the second monitor (and the group on the second monitor to switch to
the first), this command will let you do that.

### CODE REVIEW

**Comment by kovasap** [comment]
See https://groups.google.com/forum/#!msg/qtile-dev/MnyO4rMo_M0/-FN4g-gCBAAJ

**Comment by ramnes** [comment]
Great, thanks! I would actually use this. Should we add an argument or a second function to swap in the opposite direction?

**Comment by kovasap** [comment]
Added argument!  Note that I have not tested this on more than 2 monitors, so I'm not sure if the rotating behavior will work exactly as expected.

**Comment by kaistian** [comment]
I have tested on 3 monitors and it rotates perfectly both ways.

**Comment by tych0** [comment]
Can you add a test for this to make sure it doesn't break?

**Comment by kovasap** [comment]
Is there a place where such a test should go?  I'm not very familiar with
the codebase yet :(

**Comment by tych0** [comment]
Yeah, somewhere sensible in the test/ directory.

**Comment by pawamoy** [comment]
`test/test_manager.py` seems to be the right place to add a test.

**Comment by kovasap** [comment]
I'm not sure I quite understand how to access the screens on the qtile object in my test to verify everything is working.  Anyone have any pointers?

**Comment by kovasap** [comment]
Yep, that's where my latest code change is (I copied another test).  I'm
still a bit stuck tho.  Maybe I need to read some docs on the qtile object?

**Comment by kovasap** [comment]
Hey sorry for the slowness, I'm still not quite sure what's wrong with my particular test.  If there's anyone who could take a look I would really appreciate it : )

File: qtile-venv-entry [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Type=Application
+# Keywords=wm;tiling
+# ... to ...
+# /usr/share/xsessions/qtile-venv.desktop
+
+
+source ~/.bashrc
+source ~/.xinitrc
+source ~/.virtualenvs/qtile/bin/activate
+python ~/qtile/bin/qtile $*
```
Comment (ramnes) [inline qtile-venv-entry:17]: Did you really mean to add this here? Doesn't seem related to that PR.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 cairocffi >= 0.9.0
 cffi >= 1.1.0
 xcffib >= 0.10.1
+iwlib
```
Comment (ramnes) [inline requirements.txt:4]: Same here.

**Comment by ramnes** [comment]
Can you rebase on master? The diff is hard to understand with your latest push.

**Comment by kovasap** [comment]
Sorry I'm trying to fix another problem with my qtile that I ran into by
updating my fork.  I didn't mean to pollute this PR!  Still not a git
expert.  Let me see if I can fix...

**Comment by kovasap** [comment]
Continued in https://github.com/qtile/qtile/pull/2381.
